lib/friendly_shipping/services/usps_international/parse_package_rate.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module FriendlyShipping
+  module Services
+    class UspsInternational
+      class ParsePackageRate
+        # USPS returns all the info about a rate in a long string with a bit of gibberish.
+        ESCAPING_AND_SYMBOLS = /&lt;\S*&gt;/.freeze
+
+        # At the beginning of the long String, USPS keeps a copy of its own name. We know we're dealing with
+        # them though, so we can filter that out, too.
+        LEADING_USPS = /^USPS /.freeze
+
+        # This combines all the things we want to filter out.
+        SERVICE_NAME_SUBSTITUTIONS = /#{ESCAPING_AND_SYMBOLS}|#{LEADING_USPS}/.freeze
+
+        # Often we get a multitude of rates for the same service given some combination of
+        # Box type and (see below) and "Hold for Pickup" service. This creates a regular expression
+        # with groups named after the keys from the `Usps::CONTAINERS` constant.
+        # Unfortunately, the keys don't correspond directly to the codes we use when serializing the
+        # request.
+        BOX_REGEX = {
+          flat_rate_boxes: 'Flat Rate Box',
+          large_flat_rate_box: 'Large Flat Rate Box',
+          medium_flat_rate_box: 'Medium Flat Rate Box',
+          small_flat_rate_box: 'Small Flat Rate Box',
+          flat_rate_envelope: 'Flat Rate Envelope',
+          legal_flat_rate_envelope: 'Legal Flat Rate Envelope',
+          padded_flat_rate_envelope: 'Padded Flat Rate Envelope',
+          gift_card_flat_rate_envelope: 'Gift Card Flat Rate Envelope',
+          window_flat_rate_envelope: 'Window Flat Rate Envelope',
+          small_flat_rate_envelope: 'Small Flat Rate Envelope',
+          large_envelope: 'Large Envelope',
+          parcel: 'Parcel',
+          postcards: 'Postcards'
+        }.map { |k, v| "(?<#{k}>#{v})" }.join("|").freeze
```

Comment:
@pelargir this is more or less me imitating what i saw in Usps Domestic. Not really sure what the value is with the code here. I did pair down to what i believe is valid for international.